picker_wrapper.go

```diff
@@ -73,18 +72,17 @@ func (bp *pickerWrapper) connectionError() error {
 }
 
 func (bp *pickerWrapper) updateStickinessMDKey(newKey string) {
-	oldKeyUnsafePtr := atomic.SwapPointer(&bp.stickinessMDKey, unsafe.Pointer(&newKey))
-	if ptr := (*string)(oldKeyUnsafePtr); ptr == nil || *ptr != newKey {
+	// No need to check ok because mdKey == "" if ok == false.
+	if oldKey, _ := bp.stickinessMDKey.Load().(string); oldKey != newKey {
+		bp.stickinessMDKey.Store(newKey)
 		bp.stickiness.reset(newKey)
 	}
 }
 
 func (bp *pickerWrapper) getStickinessMDKey() string {
-	mdKeyPtr := (*string)(atomic.LoadPointer(&bp.stickinessMDKey))
-	if mdKeyPtr != nil {
-		return *mdKeyPtr
-	}
-	return ""
+	// No need to check ok because mdKey == "" if ok == false.
+	mdKey, _ := bp.stickinessMDKey.Load().(string)
```

Comment:
`return bp.stickinessMDKey.Load().(string)`

Reply:
It could be `nil`. We need `, _` otherwise it may panic.